packages/core-flows/src/definition/cart/workflows/add-shipping-method-to-cart.ts

```diff
@@ -24,22 +26,38 @@ export const addShippingMethodToWorkflow = createWorkflow(
   (
     input: WorkflowData<AddShippingMethodToCartWorkflowInput>
   ): WorkflowData<void> => {
+    const cart = useRemoteQueryStep({
+      entry_point: "cart",
+      fields: [
+        "region_id",
+        "currency_code",
+        "items.*",
+        "shipping_address.*",
+        "shipping_methods.*",
+      ],
+      variables: { id: input.cart_id },
+      list: false,
+    })
+
     const optionIds = transform({ input }, (data) => {
       return (data.input.options ?? []).map((i) => i.id)
     })
 
+    validateCartShippingOptionsStep({
```

Comment:
We need to add this validation/removal to all other cart updates to ensure that you cannot checkout with an invalid option. E.g. a customer used to have free shipping, but no longer has because they removed items from their cart.

Reply:
yes, this is coming in the next PR. I wanted to get this in separately.